### ISSUE TITLE
Cannot compile without the sixel patch

### ISSUE THREAD

**Issue by veltza** [bakkeby/st-flexipatch#118]
This is what happens when you compile st-flexipatch without the sixel patch:

```
> make
cp config.def.h config.h
cp patches.def.h patches.h
c99 -I/usr/X11R6/include  `pkg-config --cflags fontconfig`  `pkg-config --cflags freetype2`   -DVERSION=\"0.9\" -D_XOPEN_SOURCE=600  -O1 -c st.c
In file included from st.c:21:
st.c: In function ‘strhandle’:
st.c:2465:60: error: ‘osc_table’ undeclared (first use in this function)
 2465 |                         if ((j = par - 10) < 0 || j >= LEN(osc_table))
      |                                                            ^~~~~~~~~
st.h:17:41: note: in definition of macro ‘LEN’
   17 | #define LEN(a)                  (sizeof(a) / sizeof(a)[0])
      |                                         ^
st.c:2465:60: note: each undeclared identifier is reported only once for each function it appears in
 2465 |                         if ((j = par - 10) < 0 || j >= LEN(osc_table))
      |                                                            ^~~~~~~~~
st.h:17:41: note: in definition of macro ‘LEN’
   17 | #define LEN(a)                  (sizeof(a) / sizeof(a)[0])
      |                                         ^
make: *** [Makefile:19: st.o] Error 1
```

The reason is that `osc_table` is inside a sixel block when it shouldn't be inside any block.

---

I also have another issue related to this [commit](https://github.com/bakkeby/st-flexipatch/commit/400aa4492f28ac6556727aac225bba51decc82b5). I never applied that upstream patch to st-sx because it broke my shell prompts (bash and zsh) when I pasted emojis from the clipboard into the prompt.

Here is how to reproduce it:

1. Compile st-flexipatch without patches and start it.
2. Type `echo `
3. Paste an emoji from the clipboard into the prompt five times. (You should see how strangely the prompt behaves)
4. Press enter to see that the command works correctly even though the prompt is messed up.

![issue](https://github.com/bakkeby/st-flexipatch/assets/106755522/b80a4878-5c45-4556-b88e-e8645192fe92)

Can you confirm this behavior or is there something wrong with my setup?

**Comment by bakkeby** [comment]
Yes could reproduce that issue, reverted the commit, thanks.

**Comment by veltza** [comment]
Oh boy, what the hell were they thinking when they approved that breaking patch?

**Comment by bakkeby** [comment]
It was getting pretty late yesterday, misplacing the `osc_table` was rather sloppy :)

It is pretty minor but I wanted to mention that while going through the many changes you have made in your build I spotted some curly braces missing here:
https://github.com/veltza/st-sx/blob/efe8286760dfb7171f58d69b43e39c4ab56f595c/st.c#L2915-L2919

Introduced via this commit:
https://github.com/veltza/st-sx/commit/98ced789e4e5a6c5473c1fb42434e92fc0c0ea3b

**Comment by veltza** [comment]
Thanks. Apparently I'm good at finding other people's bugs, but I can't find my own. lol

The reason I wanted to support both OSC sequence terminators (BEL and ST) was to maintain backwards compatibility and support modern applications like [vv](https://github.com/hackerb9/vv) that expect the OSC color response to end with ST. So when either terminator is used in the request, the response uses the same one. Some terminals like xterm and XFCE terminal also work this way, while others like WezTerm always use ST in the response. The vanilla st uses BEL.

**Comment by bakkeby** [comment]
You are right, I shouldn't be doing these changes so late in the evening.

I got confused because both the ligatures and the wide glyphs patches change the signature of `xdrawglyphfontspecs`. Commited some fixes for that. I was able to spawn terminals with combinations of ligatures, wide glyphs and vertcenter so let me know if you still get the segfault when enabling ligatures.

**Comment by bakkeby** [comment]
Thanks for catching that. xresources is obviously one I had enabled.
